### PR TITLE
Check email aliases before allowing a user sign-up. (Fixes #530)

### LINKED ISSUE #530: Validate against existing aliases when creating a username
We were able to sign up with a username that was already with someone's alias. After paddle subscription and landing back on the accounts page, we got a "Could not connect to Thundermail, please try again later".

Image

### PR DESCRIPTION
**Pull request by MelissaAutumn**
Fixes #530 

Easy fix! We still check for user, because recovery email is a weird edge-case where right now it can be a thundermail address. In the future we can drop the user check for just the Email check.

### CODE REVIEW

File: src/thunderbird_accounts/mail/tests/test_views.py [hunk 1]
```diff
@@ -133,6 +133,26 @@ def test_reserved(self):
...
+            mock.save_email_addresses = instance
+            response = self.client.post(
+                email_alias_url,
+                data={'email-alias': 'admin', 'domain': settings.PRIMARY_EMAIL_DOMAIN},
+                content_type='application/json',
+            )
+            self.assertEqual(response.status_code, 403)
+            self.assertEqual(
+                json.loads(response.content.decode()),
+                {'success': False, 'error': _('You cannot use this email address.')},
```
[the remark this answers is not in the document]
Comment (MelissaAutumn) [reply src/thunderbird_accounts/mail/tests/test_views.py]: Lol the allow list check is right before the email check. Honestly I feel like this is fine? Maybe @kewisch or @malini can weigh in on if we should show the "this email is already in use" error, or just redirect them.
